chalice/package.py

```diff
@@ -357,15 +386,255 @@ def _register_cfn_resource_name(self, name):
         return cfn_name
 
 
+class TerraformGenerator(TemplateGenerator):
+
+    template_file = "chalice.tf.json"
+
+    def generate(self, resources):
+        # type: (List[models.Model]) -> Dict[str, Any]
+        template = {
+            'resource': {},
+            'terraform': {
+                'required_version': '> 0.11.0'
```

Comment:
Yeah let's put a minimum pinning on this as well. It seems also worthwhile to put a major version ceiling on the provider as well. It looks like they are coming out with a major version 3 and we should be careful that it does not break the template we generate.

Reply:
there's nothing material to chalice usage in the version 3 release notes (https://www.terraform.io/docs/providers/aws/guides/version-3-upgrade.html) . ceiling pins cause more problems for end users in my experience then benefits unless their done with known cause. terraform is commonly shared ci platform tool install, so this can be quite tedious for end users since they may not control the upgrade schedule or ability to downgrade do to ceiling pins. doing a minimum version pin for what chalice generated output actually needs is good. afaics for the aws provider anything over > 2.0 should be sufficient.

Reply:
added in a aws provider > 2.0 here.

Reply:
The concern is that we don't know that for certain until we can test against version 3 (my read on the linked docs suggest the full set of changes aren't finalized yet).  And, in theory this applies to *any* future major version of this provider going forward.

While I understand version caps can be difficult, I feel that capping to an entire major version seems like a reasonable tradeoff (vs. say a minor version pin).  The expectation is that major versions will have breaking changes, so I think waiting until we can confirm we're not affected seems like a reasonable approach.

I would say this applies to anywhere we're specifying versions.